pkg/lint/rules/template.go

```diff
@@ -206,11 +206,18 @@ func validateMetadataName(obj *K8sYamlStruct) error {
 	if len(obj.Metadata.Name) == 0 || len(obj.Metadata.Name) > 253 {
 		return fmt.Errorf("object name must be between 0 and 253 characters: %q", obj.Metadata.Name)
 	}
-	// This will return an error if the characters do not abide by the standard OR if the
-	// name is left empty.
-	if err := chartutil.ValidateMetadataName(obj.Metadata.Name); err != nil {
-		return errors.Wrapf(err, "object name does not conform to Kubernetes naming requirements: %q", obj.Metadata.Name)
+	if obj.APIVersion == "rbac.authorization.k8s.io/v1" {
```

Comment:
I think the solution to solve the bug by hard coding just rbac kind is not a good idea.
All the object under Kubernetes should be checked and helm [version_skew](https://helm.sh/docs/topics/version_skew/#supported-version-skew) is also another concern which should be token into consideration.

Reply:
Do you have better suggestions to handle this kind of verification? RBAC is indeed inconsistent with other checks

Reply:
Do not check the object type. Instead call the object's validation methods against the Kubernetes object schema. Kubernetes exposes these schema validation libraries as part of [apimachinery](https://pkg.go.dev/k8s.io/apimachinery). [ValidateObjectMeta](https://pkg.go.dev/k8s.io/apimachinery@v0.20.2/pkg/api/validation#ValidateObjectMeta) might be usefu for examplel.

I haven't looked much further, but I would suggest going that route rather than hardcoding APIVersion checks.